src/PrestaShopBundle/Form/Admin/Sell/Product/FooterType.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Sell\Product;
+
+use PrestaShop\PrestaShop\Adapter\Shop\Url\ProductProvider;
+use PrestaShopBundle\Form\Admin\Type\IconButtonType;
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Form\Extension\Core\Type\ButtonType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class FooterType extends TranslatorAwareType
+{
+    /**
+     * @var ProductProvider
+     */
+    private $productUrlProvider;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    public function __construct(
```

Comment:
We use a FormType to control the footer action buttons?

I might be out of scope but, for me 
- a form type should be used when we want user to input some data then submit
- here we have no submit, when we click a button we immediately trigger

Reply:
Actually this form type does contain:
- a form input (the active one to handle Offline/Online)
- and the submit button

Those two fields, at least, do make sense in this FormType. About the other buttons I admit I hesitated, I wasn't sure if I should include them in the form type or manually add them inside this form type. But they are still action buttons related to the product (even if they are mostly handled by ajax), so it's not completely out of scope.

And the main reason I did this was to allow module integrate their own buttons in the footer simply by adding them to this form type. Also keep in mind that some buttons are still missing here:
- Add new product
- Go to catalog
- Duplicate

Which makes me wonder if I should add them right away? Since all these features and links are already accessible? (at least via legacy controller)

Reply:
Ok I just added the missing buttons actually, with associated icons It looks much better than in the previous page I think:

![Capture d’écran 2021-05-06 à 13 04 46](https://user-images.githubusercontent.com/13801017/117288328-a9e4a600-ae6b-11eb-8c24-6b56112ceb49.png)

Reply:
> a form input (the active one to handle Offline/Online)

Then it is a good thing to use a FormType 👍